gnosis/eth/clients/ens_client.py

```diff
@@ -14,15 +14,33 @@ class EnsClient:
     Resolves Ethereum Name Service domains using ``thegraph`` API
     """
 
-    def __init__(self, network_id: int):
-        self.ethereum_network = EthereumNetwork(network_id)
-        if network_id == self.ethereum_network.SEPOLIA:
-            url = (
-                "https://api.studio.thegraph.com/proxy/49574/enssepolia/version/latest/"
-            )
-        else:  # Fallback to mainnet
-            url = "https://api.thegraph.com/subgraphs/name/ensdomains/ens/"
-        self.url = url
+    class Config:
+        def __init__(self, network: EthereumNetwork, base_url: str) -> None:
+            self.network = network
+            self.base_url = base_url
+
+        @property
+        def url(self) -> str:
+            return self.base_url
+
+    class SubgraphConfig(Config):
+        def __init__(
+            self,
+            network: EthereumNetwork,
+            base_url: str,
+            api_key: str,
+            subgraph_id: str,
+        ) -> None:
+            super().__init__(network, base_url)
+            self.api_key = api_key
```

Comment:
Is the `api_key` used anywhere?

Reply:
Good catch! It was missing from the URL (should be in the path).

Updated in [877e9d6](https://github.com/safe-global/safe-eth-py/pull/1187/commits/877e9d69659eae8785b56be9fa19e68884208a59)